encodings/runend/src/compute.rs

```diff
@@ -142,11 +149,58 @@ impl SliceFn for RunEndArray {
     }
 }
 
+impl FilterFn for RunEndArray {
+    fn filter(&self, predicate: &ArrayData) -> VortexResult<ArrayData> {
+        let primitive_run_ends = self.ends().into_primitive()?;
+        let (run_ends, pred) = match_each_unsigned_integer_ptype!(primitive_run_ends.ptype(), |$P| {
+            filter_run_ends(primitive_run_ends.maybe_null_slice::<$P>(), predicate)?
+        });
+        let values = filter(self.values(), &pred)?;
+        let validity = self.validity().filter(predicate)?;
+
+        RunEndArray::try_new(run_ends.into_array(), values, validity).map(|a| a.into_array())
+    }
+}
+
+// Code adapted from apache arrow-rs https://github.com/apache/arrow-rs/blob/b1f5c250ebb6c1252b4e7c51d15b8e77f4c361fa/arrow-select/src/filter.rs#L425
```

Comment:
When you say adapted, why not just call it instead? Can construct an Arrow RunEndArray and call filter on it?

Reply:
It avoids having to canonicalize values array

Reply:
Ah gotcha 👍